backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java

```diff
@@ -374,6 +380,32 @@ public void deleteUserByEmail(String email) {
         userDao.delete(existingUser);
     }
 
+    @Override
+    public List<User> getTopUsers(String symbol) {
+        List<User> users = new ArrayList<>();
+        // get a list of all user who own a stock with this symbol
+        ownsDao.findAll().stream()
+                .filter(owns -> owns.getStock().getSymbol().equals(symbol))
+                .collect(toList())
+                .forEach(owns -> users.add(owns.getUser()));
+
+        if (users.isEmpty()) {
+            return emptyList();
+        }
+
+        // sort user by ascending score order
+        Collections.sort(users);
+
+        // reverse to get biggest score first
+        Collections.reverse(users);
```

Comment:
Everything here is fine and works. I'd just like to mention that it could be reduced:
- Instead of the .forEach after the stream, you can use .map(Owns::getUser) after the .filter
- You can then sort and reverse inside the stream, since you are now dealing with the users, as follows:
```java
.sorted(Comparator.comparing(User::getScore).reversed())
```
And since you made the User class comparable on score, you can drop the getScore

Reply:
Thks for the optimization, will add it